control-plane/agents/src/bin/core/controller/io_engine/types.rs

```diff
@@ -32,3 +32,51 @@ pub(crate) struct RebuildHistoryResp {
     pub(crate) end_time: Option<prost_types::Timestamp>,
     pub(crate) histories: HashMap<NexusId, RebuildHistory>,
 }
+
+/// Create a Snapshot Rebuild on target node where the replica resides on.
+pub(crate) struct CreateSnapRebuild {
+    /// The uuid of the rebuilding replica.
+    pub(crate) replica_uuid: transport::ReplicaId,
+    /// The nvmf URI of the source Snapshot.
+    pub(crate) source_uri: String,
+}
+/// The status of the snapshot Rebuild.
+pub(crate) type SnapshotRebuildStatus = rpc::v1::snapshot_rebuild::RebuildStatus;
+#[derive(Debug)]
+#[allow(dead_code)]
+pub(crate) struct SnapshotRebuild {
+    /// The uuid of the rebuilding replica.
+    pub(crate) uuid: transport::ReplicaId,
+    /// The nvmf URI of the source Snapshot from which we:
+    /// 1. read allocated data bitmap
+    /// 2. read data proper
+    pub(crate) source_uri: String,
+    /// Current status of the rebuild.
+    pub(crate) status: SnapshotRebuildStatus,
+    /// Total bytes to rebuild.
+    pub(crate) total: u64,
+    /// Total bytes rebuilt so far.
+    pub(crate) rebuilt: u64,
+    /// Remaining bytes to rebuild.
+    pub(crate) remaining: u64,
+    /// Cluster index of the last persistence checkpoint.
+    /// All previous clusters have been rebuilt and this
+    /// is persisted in the replica metadata.
+    pub(crate) persisted_checkpoint: u64,
+    /// Timestamp taken at the start of the rebuild.
+    pub(crate) start_timestamp: Option<SystemTime>,
+    /// Timestamp taken at the end of the rebuild.
+    pub(crate) end_timestamp: Option<SystemTime>,
+}
+/// List all or a specific snapshot rebuild.
+pub(crate) struct ListSnapRebuild {
+    /// Select a specific target replica.
+    pub(crate) uuid: Option<transport::ReplicaId>,
+}
+/// A list of snapshot rebuilds.
+pub(crate) type ListSnapRebuildRsp = Vec<SnapshotRebuild>;
+/// Destroy a snapshot rebuild.
+pub(crate) struct DestroySnapRebuild {
+    /// The uuid of the particular rebuild to destroy.
```

Comment:
```suggestion
    /// The uuid of the particular replica for which to destroy the rebuild.
```

Reply:
Ah though this is also the uuid of  the rebuild itself, which inherits its uuid from the replica. I'll also add this info in the doc comment